subprojects/instant-execution/src/main/kotlin/org/gradle/instantexecution/serialization/codecs/TaskNodeCodec.kt

```diff
@@ -144,6 +150,30 @@ class TaskNodeCodec(
         }
     }
 
+    private
+    suspend fun WriteContext.writeDestroyablesOf(task: TaskInternal) {
+        writeCollection((task.destroyables as TaskDestroyablesInternal).registeredPaths)
```

Comment:
This should be converted into a `FileCollection` and then written as such, rather than writing this as arbitrary user values. This way, we correctly evaluate and resolve user code to paths. We did the same thing for `CopySpec` and `SourceTask`. We should do this by changing `registeredPaths` to return a `FileCollection`, as all of the users of this method should be doing the same thing.

Reply:
Done in https://github.com/gradle/gradle/pull/13063